### DISCUSSION TITLE
Some types of AI text generation chat implementation.

### DISCUSSION THREAD

**Discussion by Saraivinha1703** [vercel/community#6587] [Show & Tell]
Hello 😁! Just wanted to share my repository which have some different types of AI chats using Netx.js, Vercel's [AI SDK](https://sdk.vercel.ai/docs) and some other libs. The project includes:
- a simple chat with `Hugging Face Inference` using `OpenAssistant/oasst-sft-4-pythia-12b-epoch-3.5` model
- a Chat with tools with OpenAI's `gpt-3.5-turbo-0125` <i>(fake data)</i>
- a Generative UI with Together AI as the provider using Mistral's model `mistralai/Mixtral-8x7B-Instruct-v0.1` <i>(fake data)</i>
- a simple RAG (Retrieval Augmented Generation) QA using OpenAI embeddings and chat and LangChain to talk about dividends.
- a RAG chat with the chat history and vector store context. <i>(currently working on that)</i>

The goal in my project is to implement all types of text generation chats that I can, making them as cheap as possible which means I still need to work on alternatives to OpenAI tools and RAG chats 😔, but the cheapest alternative in terms of cost and benefit that I've found so far is [Together AI](https://www.together.ai/pricing). You pay for what you use according to the price of the model and have 25 dolars to test it. I hope this helps someone, any suggestions on how to improve this project are welcome 🥳, **check out my repository [here](https://github.com/Saraivinha1703/nextjs-ai-chat)**!

**Comment by amyegan** [comment]
Thanks for sharing your work! 😄
